File: java/parser/src/main/java/gov/lbl/parser/parser/For_statementVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Statement;
import gov.lbl.parser.domain.For_statement;
import gov.lbl.parser.domain.For_indices;

import static java.util.stream.Collectors.toList;

public class For_statementVisitor extends modelicaBaseVisitor<For_statement> {
    @Override
    public For_statement visitFor_statement(modelicaParser.For_statementContext ctx) {
        For_indicesVisitor for_indicesVisitor = new For_indicesVisitor();
        For_indices for_indices = ctx.for_indices() == null? null : ctx.for_indices().accept(for_indicesVisitor);

        StatementVisitor statementVisitor = new StatementVisitor();
        List<Statement> loop_statements = ctx.statement() == null ? null : ctx.statement()
                                                                                .stream()
                                                                                .map(stmt -> stmt.accept(statementVisitor))
                                                                                .collect(toList());
        return new For_statement(for_indices, loop_statements);
    }
}

File: jsParser/parser/For_statementVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const For_statement = require('../domain/For_statement')

const StatementVisitor = require('./StatementVisitor')
const For_indicesVisitor = require('./For_indicesVisitor')

class For_statementVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitFor_statement (ctx) {
    let for_indices = null
    const loop_statements = []

    if (ctx.for_indices()) {
      const for_indicesVisitor = new For_indicesVisitor.For_indicesVisitor()
      for_indices = for_indicesVisitor.visitFor_indices(ctx.for_indices())
    }

    if (ctx.statement()) {
      const statementVisitor = new StatementVisitor.StatementVisitor()
      ctx.statement().forEach(stmt => {
        loop_statements.push(statementVisitor.visitStatement(stmt))
      })
    }

    return new For_statement.For_statement(for_indices, loop_statements)
  }
};

For_statementVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitFor_statement = this.visitFor_statement
exports.For_statementVisitor = For_statementVisitor
